internal/topology/variables/clusterclass_variable_validation.go

```diff
@@ -193,6 +149,48 @@ func validateSchema(schema *clusterv1.JSONSchemaProps, fldPath *field.Path) fiel
 	}); err != nil {
 		return append(allErrs, field.Invalid(fldPath, "", fmt.Sprintf("failed to build validator: %v", err)))
 	}
+	allErrs = append(allErrs, validateSchema(apiExtensionsSchema, fldPath)...)
+	return allErrs
+}
+
+func validateSchema(schema *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	// Validate that type is one of the validVariableTypes.
+	switch {
+	case len(schema.Type) == 0:
+		return field.ErrorList{field.Required(fldPath.Child("type"), "openAPIV3Schema.type cannot be empty")}
```

Comment:
```suggestion
		return field.ErrorList{field.Required(fldPath.Child("type"), "type cannot be empty")}
```
Let's drop it here too, given that it's wrong on deeper layers